samples/nrf9160/aws_fota/src/main.c

```diff
@@ -0,0 +1,571 @@
+#endif /* defined(CONFIG_BSD_LIBRARY) */
+#else
+	memcpy(id, CONFIG_CLOUD_CLIENT_ID, CLOUD_CLIENT_ID_LEN + 1);
+#endif /* !defined(NRF_CLOUD_CLIENT_ID) */
+
+	return 0;
+}
+
+/**@brief Initialize the MQTT client structure
+*/
+static int client_init(struct mqtt_client *client)
+{
+	mqtt_client_init(client);
+	provision();
+	broker_init();
+
+	int ret = client_id_get(client_id_buf);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* MQTT client configuration */
+	client->broker = &broker;
+	client->evt_cb = mqtt_evt_handler;
+	client->client_id.utf8 = client_id_buf;
+	client->client_id.size = CLIENT_ID_LEN;
+	client->password = NULL;
+	client->user_name = NULL;
+	client->protocol_version = MQTT_VERSION_3_1_1;
+
+	/* MQTT buffers configuration */
+	client->rx_buf = rx_buffer;
+	client->rx_buf_size = sizeof(rx_buffer);
+	client->tx_buf = tx_buffer;
+	client->tx_buf_size = sizeof(tx_buffer);
+
+	/* MQTT transport configuration */
+	client->transport.type = MQTT_TRANSPORT_SECURE;
+
+	static sec_tag_t sec_tag_list[] = {CONFIG_NRF_CLOUD_SEC_TAG};
+	struct mqtt_sec_config *tls_config = &(client->transport).tls.config;
+
+	tls_config->peer_verify = 2;
+	tls_config->cipher_list = NULL;
+	tls_config->cipher_count = 0;
+	tls_config->sec_tag_count = ARRAY_SIZE(sec_tag_list);
+	tls_config->sec_tag_list = sec_tag_list;
+	tls_config->hostname = CONFIG_AWS_HOSTNAME;
+
+	return 0;
+
+}
+
+/**@brief Initialize the file descriptor structure used by poll.
+*/
+static int fds_init(struct mqtt_client *c)
+{
+	if (c->transport.type == MQTT_TRANSPORT_NON_SECURE) {
+		fds.fd = c->transport.tcp.sock;
+	} else {
+#if defined(CONFIG_MQTT_LIB_TLS)
+		fds.fd = c->transport.tls.sock;
+#else
+		return -ENOTSUP;
+#endif
+	}
+
+	fds.events = POLLIN;
+
+	return 0;
+}
+
+/**@brief Configures modem to provide LTE link. Blocks until link is
+ * successfully established.
+ */
+static void modem_configure(void)
+{
+#if defined(CONFIG_LTE_LINK_CONTROL)
+	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
+		/* Do nothing, modem is already turned on
+		 * and connected.
+		 */
+	} else {
+		int err;
+
+		printk("LTE Link Connecting ...\n");
+		err = lte_lc_init_and_connect();
+		__ASSERT(err == 0, "LTE link could not be established.");
+		printk("LTE Link Connected!\n");
+	}
+#endif
+}
+
+void aws_fota_cb_handler(enum aws_fota_evt_id evt)
+{
+	/* When the applicaiton is ready restart */
+	return 0;
+}
+
+void main(void)
+{
+	int err;
+
+	/* The mqtt client struct */
+	struct mqtt_client client;
+
+	printk("The MQTT fota\n");
+
+	modem_configure();
+
+	client_init(&client);
+	err = aws_fota_init(&client, "v1.0.1", aws_fota_cb_handler);
+	if (err != 0) {
+		printk("ERROR: aws_fota_init %d\n", err);
+		return;
+	}
+
+	err = mqtt_connect(&client);
+	if (err != 0) {
+		printk("ERROR: mqtt_connect %d\n", err);
+		return;
+	}
+
+	err = fds_init(&client);
+	if (err != 0) {
+		printk("ERROR: fds_init %d\n", err);
+		return;
+	}
+
+	while (1) {
+		err = poll(&fds, 1, K_SECONDS(CONFIG_MQTT_KEEPALIVE));
```

Comment:
A sentence or two about what the flow in this loop is would be welcome for those not familiar with poll (like me :D).

Reply:
Should be updated in both samples will ignroe this for now.